src/freenet/client/FetchContext.java

```diff
@@ -404,6 +413,11 @@ public FetchContext(DataInputStream dis) throws StorageFormatException, IOExcept
         cooldownRetries = dis.readInt();
         cooldownTime = dis.readLong();
         ignoreUSKDatehints = dis.readBoolean();
+        s = dis.readUTF();
```

Comment:
I don’t really understand this — does it need additional checks beyond what is done for the other fields?

Reply:
In ClientGet this is read via the checksumReaderWithLength — if the length is the length of a single object, then this is safe.

Reply:
Okay, did you check that we always use the correct length? Is it hardcoded? Does it need to be adjusted? Or is the correct length otherwise known beforehand?

Reply:
This is used in a PrependLengthOutputStream which skips 8 bytes, so that’s the maximum size. As far as I understood the implementation, the size is written on closing.

But it might be an old version of the datastructure, and then this would throw a EOFException. I now added handling for that (if it doesn’t have a field for schemeHostAndPort, then the field is not set).